llvm/lib/Transforms/InstCombine/InstCombinePHI.cpp

```diff
@@ -1625,6 +1625,25 @@ Instruction *InstCombinerImpl::visitPHINode(PHINode &PN) {
     return replaceInstUsesWith(PN, &IdenticalPN);
   }
 
+  if (PN.getType()->isIntegerTy()) {
+    bool MadeChange = false;
+    SimplifyQuery Q = SQ.getWithInstruction(&PN);
+    for (unsigned I = 0, E = PN.getNumIncomingValues(); I != E; ++I) {
+      Value *V = PN.getIncomingValue(I);
+      if (isa<ConstantInt>(V))
+        continue;
+
```

Comment:
> Compile-time overhead: +~0.2%

maybe having a check like this can help the compile time regression:

```suggestion
      if (auto *BI =
              dyn_cast<BranchInst>(PN.getIncomingBlock(I)->getTerminator()))
        if (BI->isUnconditional())
          continue;

```
think that it only is for this that this simplification is needed

Reply:
hmm one test failed when I tried this so not only for conditional branches that this simplify